src/Chain.php

```diff
@@ -84,10 +84,15 @@ class Chain implements ArrayAccess, IteratorAggregate
 
     /**
      * @param array $array
+     *
+     * @return Chain
      */
-    public function __construct(array $array = [])
+    public static function create(array $array = [])
     {
-        $this->array = $array;
+        $chain = new self();
```

Comment:
Should be new static(); I think. (http://stackoverflow.com/questions/5197300/new-self-vs-new-static)
In short, new self(); will create a ```Chain``` even if uses an extended Version like so:

```php
<?php

class MyChain extends Chain 
{
  public static function create()
  {		
    return parent::create(['im so foo']); // will return a Chain not a MyChain if self is used
  }
}
```